.github/workflows/go-e2e.yml

```
@@ -0,0 +1,112 @@
+        - {name: "Singlenode (12)",          shortname: "singlenode-12",  tsdb: true,  multi: false, pg: 12}
+        - {name: "Singlenode (13)",          shortname: "singlenode-13",  tsdb: true,  multi: false, pg: 13}
+        - {name: "Singlenode",               shortname: "singlenode-14",  tsdb: true,  multi: false, pg: 14}
+        - {name: "Without TimescaleDB",      shortname: "no-timescaledb", tsdb: false, multi: false, pg: 14}
+        # TODO (james): Skipping multinode because tests are broken for now
+        # - {name: "Multinode",                shortname: "multinode",      tsdb: true,  multi: true,  pg: 14}
+    steps:
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v3
+
+    - name: Checkout test data files
+      run: wget https://github.com/timescale/promscale-test-data/raw/main/traces-dataset.sz -O pkg/tests/testdata/traces-dataset.sz
+
+    - name: Set up Go ${{ env.golang-version }}
+      uses: actions/setup-go@v3.0.0
+      with:
+        go-version: ${{ env.golang-version }}
+      id: go
+
+    - name: Use Go module caching
+      uses: actions/cache@v2
+      with:
+        path: ~/go/pkg/mod
+        key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
+        restore-keys: |
+          ${{ runner.os }}-go-
+
+    - name: Login to GitHub Container Registry
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }}
+
+    - name: Prepare metadata PRs
+      id: metadata
+      if:  github.event_name != 'workflow_dispatch'
+      run: |
+        branch_name=$(echo ${{github.head_ref || github.ref_name}} | sed 's#/#-#')
+        possible_branch_tag=$(echo ${branch_name}-ts2-pg${{matrix.test-setups.pg}})
+        develop_branch_tag=$(echo develop-ts2-pg${{matrix.test-setups.pg}})
+        image_base="ghcr.io/timescale/dev_promscale_extension"
+        docker_image=$(./scripts/fallback-docker.sh ${image_base}:${possible_branch_tag} ${image_base}:${develop_branch_tag})
+        echo "::set-output name=docker_image::${docker_image}"
+
+    - name: Prepare metadata Workflow Dispatch
+      id: metadata_wd
+      if:  github.event_name == 'workflow_dispatch'
+      run: |
+        docker_image=$(echo ${{github.event.inputs.docker_image_prefix}}-pg${{matrix.test-setups.pg}})
+        echo "::set-output name=docker_image::${docker_image}"
+
+    - name: Test ${{ matrix.test-setups.name }}
+      env:
+        DOCKER_IMAGE: ${{ github.event_name == 'workflow_dispatch' && steps.metadata_wd.outputs.docker_image || steps.metadata.outputs.docker_image }}
+        TSDB: ${{ matrix.test-setups.tsdb }}
+        MULTI: ${{ matrix.test-setups.multi }}
+        SHORTNAME: ${{ matrix.test-setups.shortname }}
+      run: go test -race -timeout=30m ./pkg/tests/end_to_end_tests/ -use-timescaledb=$TSDB -use-multinode=$MULTI -timescale-docker-image=$DOCKER_IMAGE | tee $SHORTNAME-test-run.log  2>&1
```

```suggestion
      run: go test -race -timeout=30m ./pkg/tests/end_to_end_tests/ -use-timescaledb=$TSDB -use-multinode=$MULTI -timescale-docker-image=$DOCKER_IMAGE
```
[Logs are always available for download](https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/using-workflow-run-logs#downloading-logs), so this step seems unnecessary.